source/tests/system/nifgen_driver_api_tests.cpp

```diff
@@ -0,0 +1,402 @@
+  request.set_attribute_value(expected_value);
+  fgen::SetAttributeViReal64Response response;
+  ::grpc::Status status = GetStub()->SetAttributeViReal64(&context, request, &response);
+
+  EXPECT_EQ("", status.error_message());
+  EXPECT_TRUE(status.ok());
+  EXPECT_EQ(kfgenDriverApiSuccess, response.status());
+  ViReal64 get_attribute_value_sourcedelay = get_real64_attribute(channel_name, attribute_to_set);
+  EXPECT_EQ(expected_value, get_attribute_value_sourcedelay);
+}
+
+TEST_F(nifgenDriverApiTest, SetAttributeViBoolean_GetAttributeViBooleanReturnsSameValue)
+{
+  const char* channel_name = "0";
+  const fgen::NiFgenAttributes attribute_to_set = fgen::NiFgenAttributes::NIFGEN_ATTRIBUTE_OUTPUT_ENABLED;
+  const ViBoolean expected_value = true;
+  ::grpc::ClientContext context;
+  fgen::SetAttributeViBooleanRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  request.set_channel_name(channel_name);
+  request.set_attribute_id(attribute_to_set);
+  request.set_attribute_value(expected_value);
+  fgen::SetAttributeViBooleanResponse response;
+  ::grpc::Status status = GetStub()->SetAttributeViBoolean(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  EXPECT_EQ(kfgenDriverApiSuccess, response.status());
+  ViBoolean get_attribute_value = get_bool_attribute(channel_name, attribute_to_set);
+  EXPECT_EQ(expected_value, get_attribute_value);
+}
+
+TEST_F(nifgenDriverApiTest, SetAttributeViString_GetAttributeViStringReturnsSameValue)
+{
+  const char* channel_name = "";
+  const fgen::NiFgenAttributes attribute_to_set = fgen::NiFgenAttributes::NIFGEN_ATTRIBUTE_MARKER_EVENT_OUTPUT_TERMINAL;
+  const ViString expected_value = "sample";
+  ::grpc::ClientContext context;
+  fgen::SetAttributeViStringRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  request.set_channel_name(channel_name);
+  request.set_attribute_id(attribute_to_set);
+  request.set_attribute_value(expected_value);
+  fgen::SetAttributeViStringResponse response;
+  ::grpc::Status status = GetStub()->SetAttributeViString(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  EXPECT_EQ(kfgenDriverApiSuccess, response.status());
+  std::string get_attribute_value = get_string_attribute(channel_name, attribute_to_set);
+  EXPECT_STREQ(expected_value, get_attribute_value.c_str());
+}
+
+TEST_F(nifgenDriverApiTest, ConfigureTriggerMode_ConfiguresSuccessfully)
+{
+  const char* channel_name = "0";
+  fgen::TriggerMode expected_value = fgen::TriggerMode::TRIGGER_MODE_NIFGEN_VAL_CONTINUOUS;
+  configure_trigger_mode(channel_name, expected_value);
+
+  ViInt32 actual_output_function_value = get_int32_attribute(channel_name, fgen::NiFgenAttributes::NIFGEN_ATTRIBUTE_TRIGGER_MODE);
+  EXPECT_EQ(expected_value, actual_output_function_value);
+}
+
+TEST_F(nifgenDriverApiTest, SendSoftwareTrigger_TriggersSuccessfully)
+{
+  const char* channel_name = "0";
+  ViReal64 expected_current_level = 3.0;
+
```

Comment:
- Remove blank line
- Indentation for lines below need to be fixed.

Reply:
Done